src/main/java/uk/gov/ch/developer/docs/controller/developer/UserCallbackController.java

```diff
@@ -15,22 +18,30 @@
 import uk.gov.ch.oauth.IIdentityProvider;
 import uk.gov.ch.oauth.IOauth;
 import uk.gov.ch.oauth.OAuthToken;
+import uk.gov.ch.oauth.UserProfileResponse;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.handler.SessionHandler;
 
 @Controller
 @RequestMapping("${callback.url}")
 public class UserCallbackController {
 
     private static final Logger LOGGER = LoggerFactory.getLogger("docs.developer.ch.gov.uk");
+
     @Autowired
-    IIdentityProvider identityProvider;
+    private IIdentityProvider identityProvider;
+
     @Autowired
     private IOauth oauth;
 
+    private final WebClient webClient = WebClient.create();
```

Comment:
I think `webClient` should not be a member variable, it should be a local variable within its usages.
Each time you `create()` any previous creations will need to be disposed of, or the code will leak connection handles.

Reply:
After yesterday's s conversation we agreed this should remain here, subscriptions are disposed of not the actual web client.